### ISSUE TITLE
Different frameworks, different performances.

### ISSUE THREAD

**Issue by Magallan1229** [cihangxie/NIPS2017_adv_challenge_defense#8]
A few days ago, I tried to reproduce this via PyTorch. However, the performance was not good as that showed in the paper.  I checked my code over and over again but didn`t find any error.  Finally , I had to doubt if this discrepency is to do with ML framework. To my surprise, the parameters of PyTorch pretrained inc-v3 is not the same as those of tensorflow checkpoint, although I thought they should both come from Google. So I compared Top-1 classification accuracy of two models under the vanilla attack, and the results were:

eps 0.00784313725490196
tf_model
FGSM Test loss: 4.036023061040422,Accuracy: 0.323943661971831
FGSM with rand defense Test loss: 2.0188463489774247,Accuracy: 0.6126760563380281
torch_model
FGSM Test loss: 4.023741396380142,Accuracy: 0.3028169014084507
FGSM with rand defense Test loss: 2.0354075784414585,Accuracy: 0.6056338028169014

eps 0.0196078431372549
tf_model
FGSM Test loss: 4.1045747743525975,Accuracy: 0.28169014084507044
FGSM with rand defense Test loss: 2.466089602927087,Accuracy: 0.528169014084507
torch_model
FGSM Test loss: 4.307305727206486,Accuracy: 0.2605633802816901
FGSM with rand defense Test loss: 2.7760817568067093,Accuracy: 0.4647887323943662

eps 0.0392156862745098
tf_model
FGSM Test loss: 3.7351053701320165,Accuracy: 0.34507042253521125
FGSM with rand defense Test loss: 2.516060647830157,Accuracy: 0.5492957746478874
torch_model
FGSM Test loss: 4.066972106275424,Accuracy: 0.29577464788732394
FGSM with rand defense Test loss: 2.9954927706382644,Accuracy: 0.45774647887323944 

It seems that performance of tf_model is more like what the paper shows, Pytorch pretrained model looks more fragile especially when suffer from FGSM with larger perturbations.

**Comment by cihangxie** [comment]
Interesting to see this result. I have no idea why the model in pytorch is much more fragile than the model in tensorflow.

But one suggestion is that when performing the resizing operation, you need to choose "nearest neighbor" as indicated here https://github.com/cihangxie/NIPS2017_adv_challenge_defense/blob/master/defense.py#L113. I note other resizing options will degrade performance (but do not understand why).

**Comment by Magallan1229** [comment]
Yes , I really choose "nearest neighbor". The only difference is just where the model from.  The inconsistency was still found by others here https://github.com/pytorch/vision/issues/926.

**Comment by cihangxie** [comment]
Interesting...Maybe you can temporarily follow the tensorflow implementation here to continue your research, until pytorch fix this issue :)
